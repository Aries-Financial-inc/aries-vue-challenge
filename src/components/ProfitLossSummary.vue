<template>
  <div :class="$style.Root">
    <p><label>Max Profit:</label> {{ formatPrice(maxProfit) }}</p>
    <p><label>Max Loss:</label> {{ formatPrice(maxLoss) }}</p>
    <p><label>Break-even Points:</label> {{ breakEvenPoints.join(', ') }}</p>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

/* import models */
import { Option } from '@/models/Option'

/* import utils */
import {
  getBreakEvenPoints,
  getMaxProfit,
  getMaxLoss
} from '@/utils/OptionUtils'
import { formatPrice } from '@/utils/FormatUtils'

export default Vue.extend({
  props: {
    optionsData: {
      type: Array as () => Option[],
      required: true
    }
  },

  computed: {
    formatPrice() {
      return formatPrice
    },
    breakEvenPoints(): number[] {
      return getBreakEvenPoints(this.optionsData)
    },
    maxProfit(): number {
      return getMaxProfit(this.optionsData)
    },
    maxLoss(): number {
      return getMaxLoss(this.optionsData)
    }
  }
})
</script>

<style module>
.Root {
  text-align: center;

  label {
    font-weight: bold;
  }
}
</style>
