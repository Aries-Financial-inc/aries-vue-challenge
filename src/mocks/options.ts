export const sampleData = [
  {
    strike_price: 100,
    type: 'Call',
    bid: 10.05,
    ask: 12.04,
    position: 'Long',
    expiration_date: '2025-12-17T00:00:00Z'
  },
  {
    strike_price: 102.5,
    type: 'Call',
    bid: 12.1,
    ask: 14,
    position: 'Long',
    expiration_date: '2025-12-17T00:00:00Z'
  },
  {
    strike_price: 103,
    type: 'Put',
    bid: 14,
    ask: 15.5,
    position: 'Short',
    expiration_date: '2025-12-17T00:00:00Z'
  },
  {
    strike_price: 105,
    type: 'Put',
    bid: 16,
    ask: 18,
    position: 'Long',
    expiration_date: '2025-12-17T00:00:00Z'
  }
]
