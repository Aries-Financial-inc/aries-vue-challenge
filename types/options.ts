export interface OptionContract {
  type: "call" | "put";
  strikePrice: number;
  premium: number;
  quantity: number;
}

export interface AnalysisResult {
  priceAtExpiry: number;
  profitLoss: number;
}

export interface CombinedAnalysisResult {
  priceAtExpiry: number;
  [key: string]: number; // Index signature for dynamic keys
}

export interface OptionDetails {
  type: "call" | "put";
  strikePrice: number;
  longShort: "long" | "short";
  maxProfit: number | string;
  maxLoss: number | string;
  breakeven: number;
}
