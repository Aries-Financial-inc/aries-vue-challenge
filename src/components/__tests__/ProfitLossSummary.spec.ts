import { shallowMount } from '@vue/test-utils'
import ProfitLossSummary from '@/components/ProfitLossSummary.vue'
import { sampleData } from '@/mocks/options'

describe('ProfitLossSummary', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(ProfitLossSummary, {
      propsData: {
        optionsData: sampleData
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
