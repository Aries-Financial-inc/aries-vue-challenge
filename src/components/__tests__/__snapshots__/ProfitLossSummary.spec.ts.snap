// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfitLossSummary renders correctly 1`] = `
<div class="Root">
  <p><label>Max Profit:</label> $89.45</p>
  <p><label>Max Loss:</label> -$88.00</p>
  <p><label>Break-even Points:</label> 88, 111.045, 115.55, 117.75</p>
</div>
`;
